### PR TITLE
feat(linter) customizable pre-commit version

### PR DESCRIPTION
**Pull request by amartsinovich-dialpad**
## Summary
https://dialpad.slack.com/archives/CB42LKUJ0/p1611790160023500
### Checklist

- [ ] comments/docstrings/type hints are clear
- [ ] new tests have been written
- [ ] manual testing has passed
- [ ] architecture diagrams have been updated
- [ ] I've included any special rollback strategies above
- [ ] metrics/monitors/alerts/SLOs have been added or modified

### CODE REVIEW

**Review by vgautam-dialpad — APPROVED** [review]
❤️ awesome

File: linter/orb.yaml [hunk 1]
```diff
@@ -16,8 +16,21 @@ commands:
         description: |
           Optional alternate config file.
         type: string
+      pre_commit_version:
```
Comment (TheKevJames) [inline linter/orb.yaml:19]: nit: probably just `version`, given this is scoped as `pre-commit/pre_commit_version` right now and can't refer to anything else.

File: linter/orb.yaml [hunk 2]
```diff
@@ -16,8 +16,21 @@ commands:
...
+        description: |
+          Pre-commit package version.
+        type: string
```
Comment (TheKevJames) [inline linter/orb.yaml:22]: Looks like you might want to add `default: 'latest'`? As is, being unset would hit your second condition and attempt to `pip install pre-commit==` -- keep in mind that users can call either the `pre-commit` command (this block) _or_ the fully defined job below; both need to work!

File: linter/orb.yaml [hunk 3]
```diff
@@ -51,10 +64,16 @@ jobs:
         description: |
           The python version used to run pre-commit.
         type: string
+      pre_commit_version:
```
Comment (TheKevJames) [inline linter/orb.yaml:67]: This one is good to keep as-is, since it distinguishes from `python_version` :+1:

**Review by TheKevJames — CHANGES_REQUESTED** [review]

**Review by TheKevJames — APPROVED** [review]

**Pull request merged**